### ISSUE TITLE
MacOS 15 Xcode build Error

### ISSUE THREAD

**Issue by Bayramito** [rnmapbox/maps#3625]
## Environment
- macOS 15.0 (24A335)
- @rnmapbox/maps": "^10.1.31"
-  react-native": "0.72.14"

## Steps to reproduce
![Ekran Resmi 2024-09-17 13 21 56](https://github.com/user-attachments/assets/80019c98-52e3-4a6c-9e9a-b2bb106d6e14)

i got this error after i upgraded my xcode to 16

**Comment by 73k05** [comment]
Hi,

just got the same issue few minutes ago 🫤 I also had to upgrade `sentry` to `~5.30.0` to fix another issue mixing up package deps validation with expo

Regards
73k05

**Comment by Bayramito** [comment]
> Hi,
> 
> just got the same issue few minutes ago 🫤 I also had to upgrade `sentry` to `~5.30.0` to fix another issue mixing up package deps validation with expo
> 
> Regards 73k05

Same here. I also upgraded to latest

**Comment by predescu** [comment]
I've managed to build the app on Xcode 16 and iOS 18 with the following changes:

1. upgrade the Mapbox version in the pod file
`$RNMapboxMapsVersion = '=11.6.0'`
Previously I had $RNMapboxMapsVersion = '=11.3.0'
After this first change, I've received another error, and with the help from here:
https://stackoverflow.com/a/78633109/590626
2. adding the following code into the  target.build_configurations.each do |config| block
```
if target.name == 'BoringSSL-GRPC'
  target.source_build_phase.files.each do |file|
    if file.settings && file.settings['COMPILER_FLAGS']
      flags = file.settings['COMPILER_FLAGS'].split
      flags.reject! { |flag| flag == '-GCC_WARN_INHIBIT_ALL_WARNINGS' }
      file.settings['COMPILER_FLAGS'] = flags.join(' ')
    end
  end
end
 ```

**Comment by 73k05** [comment]
Hi @predescu 

thank you for the quick & dirty, I still have the error with the command `bun run ios` I guess you execute `pod install` in the meantime?

Regars
73k05

**Comment by chris-gaona** [comment]
I got the same error with this dependency after fixing the sentry issue. These errors occurred after upgrading my mac to xcode 16 too. I was able to resolve the issue by upgrading the mapbox version in my podfile to `$RNMapboxMapsVersion = '=11.6.0'` & then upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"`

I'm running expo 51 & react native 0.74.5

**Comment by enlivn** [comment]
> I've managed to build the app on Xcode 16 and iOS 18 with the following changes:
> ...
> 2. adding the following code into the  target.build_configurations.each do |config| block
> 
> ```
> if target.name == 'BoringSSL-GRPC'
>   target.source_build_phase.files.each do |file|
>     if file.settings && file.settings['COMPILER_FLAGS']
>       flags = file.settings['COMPILER_FLAGS'].split
>       flags.reject! { |flag| flag == '-GCC_WARN_INHIBIT_ALL_WARNINGS' }
>       file.settings['COMPILER_FLAGS'] = flags.join(' ')
>     end
>   end
> end
> ```

I guess it's pretty obvious but you need to change the name check to `if target.name == 'MapboxMaps'` so in case anyone is copy-pasta'ing without reading too carefully ;) ,

```
if target.name == 'MapboxMaps'
  target.source_build_phase.files.each do |file|
    if file.settings && file.settings['COMPILER_FLAGS']
      flags = file.settings['COMPILER_FLAGS'].split
      flags.reject! { |flag| flag == '-GCC_WARN_INHIBIT_ALL_WARNINGS' }
      file.settings['COMPILER_FLAGS'] = flags.join(' ')
    end
  end
end
```

cc @73k05 in case this was your issue.

**Comment by Bayramito** [comment]
> > I've managed to build the app on Xcode 16 and iOS 18 with the following changes:
> > ...
> > 2. adding the following code into the  target.build_configurations.each do |config| block
> > ```
> > if target.name == 'BoringSSL-GRPC'
> >   target.source_build_phase.files.each do |file|
> >     if file.settings && file.settings['COMPILER_FLAGS']
> >       flags = file.settings['COMPILER_FLAGS'].split
> >       flags.reject! { |flag| flag == '-GCC_WARN_INHIBIT_ALL_WARNINGS' }
> >       file.settings['COMPILER_FLAGS'] = flags.join(' ')
> >     end
> >   end
> > end
> > ```
> 
> I guess it's pretty obvious but you need to change the name check to `if target.name == 'MapboxMaps'` so in case anyone is copy-pasta'ing without reading too carefully ;) ,
> 
> ```
> if target.name == 'MapboxMaps'
>   target.source_build_phase.files.each do |file|
>     if file.settings && file.settings['COMPILER_FLAGS']
>       flags = file.settings['COMPILER_FLAGS'].split
>       flags.reject! { |flag| flag == '-GCC_WARN_INHIBIT_ALL_WARNINGS' }
>       file.settings['COMPILER_FLAGS'] = flags.join(' ')
>     end
>   end
> end
> ```
> 
> cc @73k05 in case this was your issue.

Unfortunately this one did not help to me

**Comment by Bayramito** [comment]
<img width="278" alt="Ekran Resmi 2024-09-18 00 04 28" src="https://github.com/user-attachments/assets/28946580-ff3b-4db6-a8a5-0d751ea17106">


Now getting these

**Comment by 73k05** [comment]
Hi @Bayramito ,

It worked for me thank you guys! Indeed I followed @chris-gaona steps:
> I got the same error with this dependency after fixing the sentry issue. These errors occurred after upgrading my mac to xcode 16 too. I was able to resolve the issue by upgrading the mapbox version in my podfile to `$RNMapboxMapsVersion = '=11.6.0'` & then upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"`
> 
> I'm running expo 51 & react native 0.74.5

@enlivn thank you, I did not need to have to add the specific `COMPILER_FLAGS` loop code to make it work BUT I had to remove the `Podfile.lock` after having this issue:
```
⚠️  Something went wrong running `pod install` in the `ios` directory.
Command `pod install` failed.
└─ Cause: CocoaPods could not find compatible versions for pod "MapboxMaps":
  In snapshot (Podfile.lock):
    MapboxMaps (= 11.0.0)
  In Podfile:
    rnmapbox-maps (from `../node_modules/@rnmapbox/maps`) was resolved to 10.1.31, which depends on
      MapboxMaps (= 11.6.0)
You have either:
 * changed the constraints of dependency `MapboxMaps` inside your development pod `rnmapbox-maps`.
   You should run `pod update MapboxMaps` to apply changes you've made.
pod install --repo-update --ansi exited with non-zero code: 1
```

**Build on iOS**
- Upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"` in `package.json`
- Upgrading the mapbox version to `$RNMapboxMapsVersion = '=11.6.0'` in `ios/Podfile`
- Deleting lock file `rm ios/Podfile.lock `
- Build & Run `bun expo run ios`

Peace
73k05

**Comment by waraness99** [comment]
> I got the same error with this dependency after fixing the sentry issue. These errors occurred after upgrading my mac to xcode 16 too. I was able to resolve the issue by upgrading the mapbox version in my podfile to `$RNMapboxMapsVersion = '=11.6.0'` & then upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"`
> 
> I'm running expo 51 & react native 0.74.5

It worked! Thanks @chris-gaona!

**Comment by jael0x** [comment]
https://github.com/rnmapbox/maps/issues/3625#issuecomment-2357615603

That worked for me, thanks @73k05 

But in my case, After the app is uploaded, shows me some warnings, should I be worried about that?

![](https://github.com/user-attachments/assets/0631680d-8a64-43dd-9aac-100ef131ac48)

**Comment by gremech** [comment]
Updating maps to v11 doesn't work for me as I also use MapboxNavigation v2 which depends on Maps v10. So I had to make a dirty change directly in `ViewAnnotationManager.swift`, to get it to at least compile until I found a solution
```swift
public var annotations: [UIView: ViewAnnotationOptions] {
    idsByView.compactMapValues { [mapboxMap] id in
        try? mapboxMap.options(forViewAnnotationWithId: id)
    } as [UIView: ViewAnnotationOptions]
}
```

**Comment by gremech** [comment]
UPD. For those who have to stay on maps v10
MapboxMaps v10.18.2 is compatible with MapboxNavigation v2.18.4 and compiles on the new xcode 16
```ruby
$RNMapboxMapsVersion = '~> 10.18.2'
```

**Comment by hot2eatorg** [comment]
> Hi @Bayramito ,
> 
> It worked for me thank you guys! Indeed I followed @chris-gaona steps:
> 
> > I got the same error with this dependency after fixing the sentry issue. These errors occurred after upgrading my mac to xcode 16 too. I was able to resolve the issue by upgrading the mapbox version in my podfile to `$RNMapboxMapsVersion = '=11.6.0'` & then upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"`
> > I'm running expo 51 & react native 0.74.5
> 
> I also upgraded macOS to `Sequoia 15.0`
> 
> @enlivn thank you, I did not need to have to add the specific `COMPILER_FLAGS` loop code to make it work **but** I had to remove the `Podfile.lock` after having this issue:
> 
> ```
> ⚠️  Something went wrong running `pod install` in the `ios` directory.
> Command `pod install` failed.
> └─ Cause: CocoaPods could not find compatible versions for pod "MapboxMaps":
>   In snapshot (Podfile.lock):
>     MapboxMaps (= 11.0.0)
>   In Podfile:
>     rnmapbox-maps (from `../node_modules/@rnmapbox/maps`) was resolved to 10.1.31, which depends on
>       MapboxMaps (= 11.6.0)
> You have either:
>  * changed the constraints of dependency `MapboxMaps` inside your development pod `rnmapbox-maps`.
>    You should run `pod update MapboxMaps` to apply changes you've made.
> pod install --repo-update --ansi exited with non-zero code: 1
> ```
> 
> **Build on iOS**
> 
> * Upgrading to the latest version of this dependency: `"@sentry/react-native": "5.30.0",` in `package.json`
> * Upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"` in `package.json`
> * Upgrading the mapbox version to `$RNMapboxMapsVersion = '=11.6.0'` in `ios/Podfile`
> * Deleting lock file `rm ios/Podfile.lock `
> * Build & Run `bun expo run ios`
> 
> Expo `50.0.17` ReactNative `0.73.6` iOS `18` Xcode `16.0` macOS `Sequoia 15.0`
> 
> Peace 73k05

I owe you a beer! For expo user, just upgrade the the `@rnmapbox/maps` by running:

`npm install @rnmapbox/maps@10.1.31`

And then set the mapbox version to `11.6.0` in you plugins app config:

```json
[
    "@rnmapbox/maps",
    {
      "RNMapboxMapsVersion": "11.6.0",
    }
]
```

**Comment by hot2eatorg** [comment]
> I got the same error with this dependency after fixing the sentry issue. These errors occurred after upgrading my mac to xcode 16 too. I was able to resolve the issue by upgrading the mapbox version in my podfile to `$RNMapboxMapsVersion = '=11.6.0'` & then upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"`
> 
> I'm running expo 51 & react native 0.74.5

I owe you a beer! For expo user, just upgrade the the `@rnmapbox/maps` by running:

`npm install @rnmapbox/maps@10.1.31`

And then set the mapbox version to `11.6.0` in you plugins app config:

```json
[
    "@rnmapbox/maps",
    {
      "RNMapboxMapsVersion": "11.6.0",
    }
]
```

**Comment by tyrauber** [comment]
> > I got the same error with this dependency after fixing the sentry issue. These errors occurred after upgrading my mac to xcode 16 too. I was able to resolve the issue by upgrading the mapbox version in my podfile to `$RNMapboxMapsVersion = '=11.6.0'` & then upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"`
> > I'm running expo 51 & react native 0.74.5
> 
> I owe you a beer! For expo user, just upgrade the the `@rnmapbox/maps` by running:
> 
> `npm install @rnmapbox/maps@10.1.31`
> 
> And then set the mapbox version to `11.6.0` in you plugins app config:
> 
> ```json
> [
>     "@rnmapbox/maps",
>     {
>       "RNMapboxMapsVersion": "11.6.0",
>     }
> ]
> ```

Thanks for this. Unfortunately, I am getting a 401 unauthorized on pod install during expo prebuild. Anyone else experiencing this?

```

[!] Error installing MapboxCommon
[!] /usr/bin/curl -f -L -o /var/folders/8f/fkvj5sln16sf4k_hzlm8r9400000gn/T/d20240926-17074-2lqtbg/file.zip https://api.mapbox.com/downloads/v2/mapbox-common/releases/ios/packages/24.6.0/MapboxCommon.zip --create-dirs --netrc-optional --retry 2 -A 'CocoaPods/1.15.2 cocoapods-downloader/2.1'

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   179    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (56) The requested URL returned error: 401

```

**Comment by enlivn** [comment]
> > [#3625 (comment)](https://github.com/rnmapbox/maps/issues/3625#issuecomment-2357615603)
> > That worked for me, thanks @73k05
> > But in my case, After the app is uploaded, shows me some warnings, should I be worried about that?
> > ![](https://private-user-images.githubusercontent.com/20349358/368756715-0631680d-8a64-43dd-9aac-100ef131ac48.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjcxMjA1MjAsIm5iZiI6MTcyNzEyMDIyMCwicGF0aCI6Ii8yMDM0OTM1OC8zNjg3NTY3MTUtMDYzMTY4MGQtOGE2NC00M2RkLTlhYWMtMTAwZWYxMzFhYzQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTIzVDE5MzcwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAyZTJhNmUxMzNlMTc4OTE4Njk2ZGM2YTE5NDkyOGQ3MTM1NTgwOTg5OWNkZGUxNmQ0ZGFjYzM3ZDFlY2U0NDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xhO3TpwF03MQYBo90UhMrIPBgqy-ZFAaurtOVvxdxbU)
> 
> @jael0x do you want to post this as a new issue? I was wondering about this too. Without the dSYMs, it's gonna be annoying/impossible to de-symbolicate stack traces once the app is released.
> 
> edit: just want to note that I've tried the following 2 things without success
> 
> 1. add the Mapbox frameworks to the `Link Binary with Libraries` in the `xcodeproj`
> 2. update the `Podfile` post-install hooks:
>    ```
>    post_install do |installer|
>      installer.pods_project.targets.each do |target|
>        target.build_configurations.each do |config|
>          config.build_settings['DEBUG_INFORMATION_FORMAT'] = 'dwarf-with-dsym'
>        end
>      end
>    end
>    ```

didn't hear back so I've created an issue myself: #3637

**Comment by aguilaair** [comment]
For those who are in v10:

```
 RNMapboxMapsVersion: "10.9.0",
```

Source: https://github.com/mapbox/mapbox-maps-ios/issues/2233

**Comment by MarchewkaMatthew** [comment]
> > I got the same error with this dependency after fixing the sentry issue. These errors occurred after upgrading my mac to xcode 16 too. I was able to resolve the issue by upgrading the mapbox version in my podfile to `$RNMapboxMapsVersion = '=11.6.0'` & then upgrading to the latest version of this dependency: `"@rnmapbox/maps": "10.1.31"`
> > I'm running expo 51 & react native 0.74.5
> 
> I owe you a beer! For expo user, just upgrade the the `@rnmapbox/maps` by running:
> 
> `npm install @rnmapbox/maps@10.1.31`
> 
> And then set the mapbox version to `11.6.0` in you plugins app config:
> 
> ```json
> [
>     "@rnmapbox/maps",
>     {
>       "RNMapboxMapsVersion": "11.6.0",
>     }
> ]
> ```

Hi @hot2eatorg, thanks for sharing the fix for Expo, could you tell me how you menaged to install the `"@rnmapbox/maps": "10.1.31"` with this version of expo and react native? I was trying to do that but I receive error described in this comment:
https://github.com/rnmapbox/maps/issues/1407#issuecomment-2411168661

**Comment by SethArchambault** [comment]
Just noting - Our project is in limbo right now.. We can't move to MapboxMapVersion 11 since that doesn't support terrain yet, and MapboxMapVersion 10.18.* has some weird glitch where the satellite view isn't fully loading in some cases. 

Hoping v10.1.32 comes out soon so we're able to deploy on ios again. Thanks!

**Comment by mfazekas** [comment]
Should be addressed by `10.1.32`

**Comment by nthh** [comment]
I ran into this issue upgrading to v11. 

macOS: 15.1.1
Xcode: 16.2
Expo SDK: ^53.0.0
@rnmapbox/maps: 10.2.9

expo plugin: 
```
   [
        "@rnmapbox/maps",
        {
          "RNMapboxMapsVersion": "11.16.3"
        }
      ]
```

What seemed to fix it while using `eas build --local` was some combination of adding the flag `--clear-cache` and running `pod repo update`.
